### PR TITLE
[CARBONDATA-1326] Fixed normal/low priority findbug issues

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3225/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/629/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3226/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/630/

File: core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java [hunk 1]
```diff
@@ -31,7 +31,9 @@
  */
 public class Blocklet implements Serializable {
 
-  private Path path;
+  private static final long serialVersionUID = 7483063116325388460L;
+
+  private String path;
```
Comment (sraghunandan) [inline core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java]: Why changed Path to string?
But constructor not changed to handle the datatype change

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java [hunk 2]
```diff
@@ -1067,7 +1071,7 @@ public static void clearBlockCache(List<AbstractIndex> dataBlocks) {
   public static byte[] packByteBufferIntoSingleByteArray(ByteBuffer[] byteBufferArr) {
     // for empty array means there is no data to remove dictionary.
     if (null == byteBufferArr || byteBufferArr.length == 0) {
-      return null;
+      return new byte[0];
```
Comment (sraghunandan) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Update the IndexKey constructor and remove null check

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java [hunk 3]
```diff
@@ -1691,7 +1695,7 @@ public static boolean isValidSortOption(String sortScope) {
     if (sortScope == null) {
       return false;
     }
-    switch (sortScope.toUpperCase()) {
+    switch (sortScope.toUpperCase(Locale.ENGLISH)) {
```
Comment (sraghunandan) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Should not use Locale.ENGLISH.Refer to String documentation in idk
use Locale.ROOT

File: hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputSplitTaskInfo.java [hunk 4]
```diff
@@ -67,6 +67,19 @@ public CarbonInputSplitTaskInfo(String taskId, List<CarbonInputSplit> carbonSpli
     return taskId.compareTo(((CarbonInputSplitTaskInfo) o).getTaskId());
   }
 
+  @Override public boolean equals(Object obj) {
+    if (this == obj) return true;
```
Comment (sraghunandan) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputSplitTaskInfo.java]: no braces

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by mohammadshahidkhan** [comment]
not required

**Pull request closed without merging**